### PR TITLE
Feature/suggest spring boot dependencies when missing

### PR DESCRIPTION
**Pull request by arik-dig**
addresses issue #859 , Sub Task 3 and 5

similar to PR #871 - Quarkus dependencies

here are the outcomes of this PR

* settings to choose Spring Boot observability mode **OtelAgent** or **Micrometer**
image

* Pop up which identifies for missing dependencies

image

and once you click you'll see in the build.gradle the following new dependencies

```
    implementation 'org.springframework.boot:spring-boot-starter-actuator:3.1.1'
    implementation 'io.micrometer:micrometer-tracing-bridge-otel:1.1.2'
    implementation 'io.opentelemetry:opentelemetry-exporter-otlp:1.26.0'
```

### CODE REVIEW

File: ide-common/src/main/java/org/digma/intellij/plugin/settings/SettingsComponent.java [hunk 1]
```diff
@@ -106,15 +107,26 @@ public boolean verify(JComponent input) {
...
         });
 
+        var mySpringBootObservabilityModeLabel = new JBLabel("Spring Boot observability mode: ");
+        mySpringBootObservabilityModeLabel.setToolTipText(""
+                + "OtelAgent will use the default OpenTelemetry java agent. "
+                + "Micrometer will use Micrometer tracing, including the annotation of 'Observed' "
+        );
+        mySpringBootObservabilityModeComboBox.addItemListener(e -> {
+            SpringBootObservabilityMode selected = (SpringBootObservabilityMode) mySpringBootObservabilityModeComboBox.getSelectedItem();
+        });
```
Comment (asafchen-dig) [inline ide-common/src/main/java/org/digma/intellij/plugin/settings/SettingsComponent.java:129]: What does it do?

**Review by shalom938 — APPROVED** [review]

File: src/main/kotlin/org/digma/intellij/plugin/ui/frameworks/SpringBootMicrometerConfigureDepsPanel.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import javax.swing.BoxLayout
+import javax.swing.JLabel
+import javax.swing.JPanel
+
+class SpringBootMicrometerConfigureDepsPanel(
+    val project: Project,
+) : DigmaResettablePanel(), Disposable {
+
+    init {
+        theService().affectedPanel = this
```
Comment (asafchen-dig) [inline src/main/kotlin/org/digma/intellij/plugin/ui/frameworks/SpringBootMicrometerConfigureDepsPanel.kt:27]: "theService()"?

**Review by asafchen-dig — APPROVED** [review]

File: ide-common/src/main/java/org/digma/intellij/plugin/settings/SettingsComponent.java (hunk 1 above)
Comment (arik-dig) [reply ide-common/src/main/java/org/digma/intellij/plugin/settings/SettingsComponent.java:129]: actually nothing... to many copy-paste :-) 
i'll remove it

File: src/main/kotlin/org/digma/intellij/plugin/ui/frameworks/SpringBootMicrometerConfigureDepsPanel.kt (hunk 2 above)
Comment (arik-dig) [reply src/main/kotlin/org/digma/intellij/plugin/ui/frameworks/SpringBootMicrometerConfigureDepsPanel.kt:27]: this view has only ONE service it works with

**Pull request merged**